web/src/user_profile.js

```diff
@@ -104,6 +106,26 @@ export function update_bot_profile_modal_ui(bot_id) {
     }
 }
 
+export function update_user_profile_modal_ui(user, new_data) {
+    if (!(modals.any_active() && modals.active_modal() === "#user-profile-modal")) {
+        return;
+    }
+    if (new_data.avatar_url !== undefined) {
+        $("#avatar").css("background-image", `url(${people.medium_avatar_url_for_person(user)})`);
+    }
+    if (new_data.delivery_email !== undefined) {
+        $("email").find(".value").text(new_data.delivery_email);
+    }
+    if (new_data.role !== undefined) {
+        $("#user-type").find(".value").text(new_data.role);
+    }
+    const user_type = {
+        name: user.full_name,
+        should_add_guest_user_indicator: people.should_add_guest_user_indicator(user.user_id),
+    };
+    $(".user_profile_name").html(render_user_full_name(user_type));
```

Comment:
Might be good to use a more specific selector here.

Reply:
Fixed it.